### PR TITLE
Feature/qr code

### PR DESCRIPTION
**Pull request by cNoveron**

### CODE REVIEW

File: apps/mobile/src/components/EditScreenInfo.tsx [hunk 1]
```diff
@@ -7,31 +7,15 @@ import { Text, View } from './Themed'
...
-
-                <View
-                    style={[styles.codeHighlightContainer, styles.homeScreenFilename]}
-                    darkColor='rgba(255,255,255,0.05)'
-                    lightColor='rgba(0,0,0,0.05)'>
-                    <MonoText>{path}</MonoText>
-                </View>
-
-                <Text style={styles.getStartedText} lightColor='rgba(0,0,0,0.8)' darkColor='rgba(255,255,255,0.8)'>
+                    Scan the QR code in your new acquaintance's mobile app:{' '}
```
Comment (asyaasha) [inline apps/mobile/src/components/EditScreenInfo.tsx:14]: I'm not sure we are meant to make changes here - I believe this is just a placeholder file meant to show message of what file you should be editing to update any particular screen ( in this case should be QR )

**Review by ottodevs — CHANGES_REQUESTED** [review]
Nice work, thanks!

There are some details that maybe we want to review, but nothing really critical.
Also the UI does not completely match the figma designs, but I guess this is not really important for now, we can work into polish those later.

I can also help fixing these details so we can merge this branch more quickly

File: apps/mobile/package.json [hunk 2]
```diff
@@ -30,6 +31,10 @@
         "react": "18.0.0",
         "react-dom": "18.0.0",
         "react-native": "0.69.5",
+        "react-native-camera": "4.2.1",
```
[the remark this answers is not in the document]
Comment (ottodevs) [reply apps/mobile/package.json]: Fixed in [4f238e9](https://github.com/Hackerthonweb3/business-card/pull/9/commits/4f238e9fe47ef3513461aa6c45c4971da7890dbf)

File: apps/mobile/package.json [hunk 3]
```diff
@@ -30,6 +31,10 @@
...
+        "react-native-paper": "4.12.4",
+        "react-native-permissions": "3.6.1",
+        "react-native-qrcode-scanner": "1.5.5",
```
[the remark this answers is not in the document]
Comment (ottodevs) [reply apps/mobile/package.json]: Fixed in [4f238e9](https://github.com/Hackerthonweb3/business-card/pull/9/commits/4f238e9fe47ef3513461aa6c45c4971da7890dbf)

File: apps/mobile/package.json [hunk 4]
```diff
@@ -30,6 +31,10 @@
         "react": "18.0.0",
         "react-dom": "18.0.0",
         "react-native": "0.69.5",
+        "react-native-camera": "4.2.1",
+        "react-native-paper": "4.12.4",
```
[the remark this answers is not in the document]
Comment (ottodevs) [reply apps/mobile/package.json]: Fixed in [4f238e9](https://github.com/Hackerthonweb3/business-card/pull/9/commits/4f238e9fe47ef3513461aa6c45c4971da7890dbf)

File: apps/mobile/package.json [hunk 5]
```diff
@@ -30,6 +31,10 @@
...
+        "react-native-permissions": "3.6.1",
```
[the remark this answers is not in the document]
Comment (ottodevs) [reply apps/mobile/package.json]: Fixed in [4f238e9](https://github.com/Hackerthonweb3/business-card/pull/9/commits/4f238e9fe47ef3513461aa6c45c4971da7890dbf)

File: apps/mobile/src/screens/Contacts.tsx [hunk 6]
```diff
@@ -12,12 +12,12 @@ export default function Contacts({ navigation }: RootTabScreenProps<'Contacts'>)
...
             <Text style={styles.title}>Contacts</Text>
             <View style={styles.content}>
-                <SurpriseIcon transform={{ scale: 1.15, x: -4 }} />
+                <SurpriseIcon /* transform={{ scale: 1.15, x: -4 }} */ />
                 <Text style={styles.heading}>You have no one yet</Text>
                 <Text style={styles.text}>
                     Have your friends scan your QR code or share your links to make a list of contacts
                 </Text>
-                <ArrowDown height={115} transform={{ scale: 1.15, y: -2 }} />
+                <ArrowDown height={115} /* transform={{ scale: 1.15, y: -2 }} */ />
```
[the remark this answers is not in the document]
Comment (ottodevs) [reply apps/mobile/src/screens/Contacts.tsx]: Fixed in [af6c8e2](https://github.com/Hackerthonweb3/business-card/pull/9/commits/af6c8e2b729b9a9417e5ab6713b734950c1a8046)

File: apps/mobile/src/screens/Contacts.tsx [hunk 7]
```diff
@@ -12,12 +12,12 @@ export default function Contacts({ navigation }: RootTabScreenProps<'Contacts'>)
         <View style={styles.container}>
             <Text style={styles.title}>Contacts</Text>
             <View style={styles.content}>
-                <SurpriseIcon transform={{ scale: 1.15, x: -4 }} />
+                <SurpriseIcon /* transform={{ scale: 1.15, x: -4 }} */ />
```
[the remark this answers is not in the document]
Comment (ottodevs) [reply apps/mobile/src/screens/Contacts.tsx]: Fixed in [af6c8e2](https://github.com/Hackerthonweb3/business-card/pull/9/commits/af6c8e2b729b9a9417e5ab6713b734950c1a8046)

File: apps/mobile/src/components/EditScreenInfo.tsx (hunk 1 above)
Comment (ottodevs) [reply apps/mobile/src/components/EditScreenInfo.tsx]: I removed the placeholder component.

File: apps/mobile/src/components/EditScreenInfo.tsx [hunk 8]
```diff
@@ -7,31 +7,15 @@ import { Text, View } from './Themed'
 
 export default function EditScreenInfo({ path }: { path: string }) {
     return (
-        <View>
+        <View style={styles.screen}>
             <View style={styles.getStartedContainer}>
+                <Text style={styles.title}>How to get a Proof of Networking</Text>
```
[the remark this answers is not in the document]
Comment (ottodevs) [reply apps/mobile/src/components/EditScreenInfo.tsx]: Removed placeholder text

File: apps/mobile/src/components/EditScreenInfo.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (ottodevs) [reply apps/mobile/src/components/EditScreenInfo.tsx]: Removed text

File: apps/mobile/src/components/EditScreenInfo.tsx [hunk 9]
```diff
@@ -7,31 +7,15 @@ import { Text, View } from './Themed'
...
-                    style={[styles.codeHighlightContainer, styles.homeScreenFilename]}
-                    darkColor='rgba(255,255,255,0.05)'
-                    lightColor='rgba(0,0,0,0.05)'>
-                    <MonoText>{path}</MonoText>
-                </View>
-
-                <Text style={styles.getStartedText} lightColor='rgba(0,0,0,0.8)' darkColor='rgba(255,255,255,0.8)'>
+                    Scan the QR code in your new acquaintance's mobile app:{' '}
+                    <MonoText style={[styles.codeHighlightContainer, styles.homeScreenFilename]}>{path}</MonoText>{' '}
                     Change any of the text, save the file, and your app will automatically update.
```
[the remark this answers is not in the document]
Comment (ottodevs) [reply apps/mobile/src/components/EditScreenInfo.tsx]: Removed the text

File: apps/mobile/src/screens/QRCode.tsx [hunk 10]
```diff
@@ -1,35 +1,265 @@
-import { StyleSheet } from 'react-native'
-
+import React, { useState, useEffect } from 'react';
+import {
+	ActivityIndicator,
+    AppRegistry,
+    Button,
+    StyleSheet,
+    // Text,
```
[the remark this answers is not in the document]
Comment (ottodevs) [reply apps/mobile/src/screens/QRCode.tsx]: Removed unused imports and dead code

File: apps/mobile/src/screens/QRCode.tsx [hunk 11]
```diff
@@ -1,35 +1,265 @@
...
+    TouchableOpacity,
+    Linking,
+    Image,
+    Modal,
+} from 'react-native';
 import EditScreenInfo from '../components/EditScreenInfo'
```
[the remark this answers is not in the document]
Comment (ottodevs) [reply apps/mobile/src/screens/QRCode.tsx]: Removed unused imports and dead code

File: apps/mobile/src/screens/QRCode.tsx [hunk 12]
```diff
@@ -1,35 +1,265 @@
...
+        setHasPermission(status === 'granted');
+      };
+  
+      getBarCodeScannerPermissions();
+    }, []);
+  
+	
+	
+    if (hasPermission === null) {
+      return <Text>Requesting for camera permission</Text>;
```
[the remark this answers is not in the document]
Comment (ottodevs) [reply apps/mobile/src/screens/QRCode.tsx]: Removed permissions management until implementing a better solution (for handling permission denied and so on)
There is now a pending task to handle this separately.

File: apps/mobile/src/screens/QRCode.tsx [hunk 13]
```diff
@@ -1,35 +1,265 @@
...
+				<Text style={styles.header} category="h4">
+					Get Proof of Networking
+				</Text>
+			</View>
+			
+            <View style={styles.container}>
+				<View style={styles.textContainer}>
+					<Text style={styles.title}>How to get a Proof of Networking</Text>
+					<Text style={styles.text} lightColor='rgba(0,0,0,0.8)' darkColor='rgba(255,255,255,0.8)'>
+						Scan the QR code in your new acquaintance's mobile app:{' '}
```
[the remark this answers is not in the document]
Comment (ottodevs) [reply apps/mobile/src/screens/QRCode.tsx]: Removed the text.

**Review by ottodevs — APPROVED** [review]

**Pull request merged**